### ISSUE TITLE
Radical Proposal: Get Rid of Record Cloning in Spine

### ISSUE THREAD

**Issue by mitchlloyd** [spine/spine#276]
TLDR: Alex, could you remove record cloning from Spine?

The behavior where finds() and other methods that return records make a clone has always been a stumbling point [for me](https://groups.google.com/forum/?fromgroups#!searchin/spinejs/mitch/spinejs/tEq0UTCOfS8/E8UTzQL8f2wJ) and [others](https://groups.google.com/forum/?fromgroups#!searchin/spinejs/peter/spinejs/IYlXjRWlLbY/NQFj6iY-WiQJ) as [well](https://github.com/maccman/spine/issues/10).  Whenever one would normally set an object as a property of some other object, one has to resort to building hash tables as a workaround.  Here are some use cases where this behavior seems unnatural:
- I would like to store a snapshot of attributes on a model to do dirty attribute tracking. One must create a hash table on the class and do lookups to accomplish this.
- I would like to store an object for an embedded document on a given document.  I must instead create a full Spine model for the embedded document and assign IDs so I can look up these models in a separate class when needed. 
- I would like to store state information that is specific to an instance of the application but should not be saved in the database.  For instance `item.queuedForDeletion = true`.  This should be visible to all controllers that rely on this object. A workaround would be to just assign another configured attribute even though this isn't saved to a saver. 

What do we loose by removing this functionality?  I'm not really sure.  To experiment I changed clone() to just return `this`.  I ran through my fairly complex application (14 distinct models, 28 controllers) and I didn't see any issues off hand.  I've read Github issues, Google Group posts, and even [Alex's Book](http://www.amazon.com/JavaScript-Web-Applications-Alex-MacCaw/dp/144930351X), but I still don't fully understand the rational behind it. Why is it so important to keep all of these copies of records unmodified until calling save?
